src/k8s/pkg/k8sd/api/component.go

```diff
@@ -11,195 +11,138 @@ import (
 	"github.com/canonical/k8s/pkg/k8sd/api/impl"
 	"github.com/canonical/k8s/pkg/snap"
 	"github.com/canonical/lxd/lxd/response"
-	"github.com/canonical/microcluster/rest"
 	"github.com/canonical/microcluster/state"
 )
 
-var k8sdComponents = rest.Endpoint{
-	Path: "k8sd/components",
-	Get:  rest.EndpointAction{Handler: componentsGet, AllowUntrusted: false},
-}
-
-var k8sdDNSComponent = rest.Endpoint{
-	Path: "k8sd/components/dns",
-	Put:  rest.EndpointAction{Handler: dnsComponentPut, AllowUntrusted: false},
-}
-
-var k8sdNetworkComponent = rest.Endpoint{
-	Path: "k8sd/components/network",
-	Put:  rest.EndpointAction{Handler: networkComponentPut, AllowUntrusted: false},
-}
-
-var k8sdStorageComponent = rest.Endpoint{
-	Path: "k8sd/components/storage",
-	Put:  rest.EndpointAction{Handler: storageComponentPut, AllowUntrusted: false},
-}
-
-var k8sdIngressComponent = rest.Endpoint{
-	Path: "k8sd/components/ingress",
-	Put:  rest.EndpointAction{Handler: ingressComponentPut, AllowUntrusted: false},
-}
-
-var k8sdGatewayComponent = rest.Endpoint{
-	Path: "k8sd/components/gateway",
-	Put:  rest.EndpointAction{Handler: gatewayComponentPut, AllowUntrusted: false},
-}
-
-func componentsGet(s *state.State, r *http.Request) response.Response {
+func getComponents(s *state.State, r *http.Request) response.Response {
 	snap := snap.SnapFromContext(s.Context)
 
 	components, err := impl.GetComponents(snap)
 	if err != nil {
-		return response.SmartError(fmt.Errorf("failed to get components: %w", err))
+		return response.InternalError(fmt.Errorf("failed to get components: %w", err))
```

Comment:
nit: SmartError will return better error codes and fallback to `InternalError` as a generic error. 
IMHO we should use `SmartError` instead of `InternalError`.

Reply:
Reasoning: I feel better declaring this explicitly, especially for 4xx vs 5xx errors (so that the client can differentiate between something he did wrong, or whether something is wrong with k8sd). For example, a file not found error would be a "500", not a "404".